closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -774,12 +778,12 @@ boolean isRestrictedServiceAllowed(Service service) {
                 String cAlgorithm = constraint.algorithm;
                 String cAttribute = constraint.attributes;
 
-                if (!isAsterisk(cType) && !type.equals(cType)) {
+                if (!isAsterisk(cType) && !type.equalsIgnoreCase(cType)) {
```

Comment:
The detailed description in a commit message should be separated from the summary line by a blank line. More information is available at https://github.com/eclipse/omr/blob/master/CONTRIBUTING.md#commit-guidelines.

Thank you for mentioning this in the commit message. Unfortunately, it isn't as helpful as I had hoped. It should also explain _why_  service types should not be case-sensitive.

Reply:
I take a look at the Java Cryptography Architecture (JCA) Reference Guide. The service algorithm name is not case-sensitive, for example, “SHA-1” and “sha-1” are the same. 

But for the service type, it is not mentioned. The “cType” in the code is got from the constraint list in java.security file, which we follow the standard name rule, for example, “MessageDigest”. And for the type which is got from the service.getType() method, it also follow the standard name rule, for example, “MessageDigest”. 

So, use “equals” or “equalsIgnoreCase” are actually same here, since we follow the standard name rule. Do you think, we still need to change to “equalsIgnoreCase”, or just leave what it is, the “equals” is good enough?

Reply:
I think this change is appropriate (see `Provider.getService(String type, String algorithm)`) where `type` is does not need to match exactly to find a given service. The benefit of this change is to extend case-insensitive matching to service types mentioned in `java.security`. However, I don't think it should be hidden in this pull request. Please revert it here and create a separate pull request. That pull request should explain the motivation in _both_ the commit message and the description.

Reply:
Ok, the change is reverted and I will create a separate PR for changing the “equals” to “equalsIgnoreCase”.